frontend/src/app/features/work-packages/components/wp-new/wp-create.service.ts

```diff
@@ -261,17 +272,25 @@ export class WorkPackageCreateService extends UntilDestroyedMixin {
    *
    * The ignoring functionality could be generalized.
    *
-   * @params object
+   * @param object
    * @param defaults
+   * @param projectIdentifier
    */
-  private defaultsFromFilters(object:HalSource|WorkPackageChangeset, defaults?:HalSource) {
+  private defaultsFromFilters(object:HalSource|WorkPackageChangeset, defaults?:HalSource, projectIdentifier?:string|null|undefined):void {
     // Not using WorkPackageViewFiltersService here as the embedded table does not load the form
     // which will result in that service having empty current filters.
     const query = this.querySpace.query.value;
 
     if (query) {
       const except:string[] = defaults?._links ? Object.keys(defaults._links) : [];
 
+      // Special case due to the introduction of the project include dropdown
+      // If we are in a project, we want the create wp to be part of that project.
+      // Only on the global WP page, the filters should be applied.
+      if (projectIdentifier) {
+        except.push('project');
```

Comment:
Did you try passing it as a default value to the create service instead (i.e., adding a project link here: https://github.com/opf/openproject/blob/dev/frontend/src/app/features/team-planner/team-planner/planner/team-planner.component.ts#L729-L732) ?

Reply:
That would work for the team planner but since the error appears on every page where the project include button exists, we'd have to do it for each of them as well. I'd like to avoid that as it is very error-prone can can easily be forgotten.